cmd/humanInsertion/humanInsertion.go

```diff
@@ -0,0 +1,81 @@
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/maf"
+	"log"
+)
+
+func humanInsertion(mafFile string, outBed_ins string, outBed_del string, species_ins string, species_del string) {
+	//initialize variables
+	mafRecords := maf.Read(mafFile) //Read entire mafFile. mafRecords has type Maf
+	var bedList_ins []*bed.Bed      //initialize 2 bed files
+	var bedList_del []*bed.Bed      //1 bed file for ins, 1 bed file for del
+
+	//go through each line
+	for i, _ := range mafRecords { //each i is a block
+		for k, _ := range mafRecords[i].Species { //each k is a line
+
+			//convert maf to bed, start with getting assembly because it is needed to verify species_ins and species_del
+			assembly_del, chrom_del := maf.SrcToAssemblyAndChrom(mafRecords[i].Species[k].Src) //start with species_del, get assembly (e.g. panTro6), chrom (e.g. chrI)
+
+			//get eC species_del lines
+			if mafRecords[i].Species[k].ELine != nil {
+				if mafRecords[i].Species[k].ELine.Status == 'C' && assembly_del == species_del { //I decided to check for both Status and Src here because they are on the same data level
+
+					//to avoid k-1 index error such as k=0,k-1=-1, only start converting maf to bed for species_ins if species_del is confirmed
+					//here I assume species_del is query sequence; species_ins is target sequence; only pairwise alignment
+					assembly_ins, chrom_ins := maf.SrcToAssemblyAndChrom(mafRecords[i].Species[k-1].Src) //then find corresponding species_ins line, same block, 1 line above since pairwise maf
+					//get corresponding s species_ins lines
+					if assembly_ins != species_ins { //verify line k-1 is indeed species_ins
+						log.Fatalf("species_ins was incorrect. Please check you have a pairwise maf file, and entered species_ins and species_del correctly") //otherwise fatal
+					}
+					if mafRecords[i].Species[k-1].SLine != nil { //if corresponding species_ins line an s line
+
+						//convert maf to bed, continued
+						current_del := bed.Bed{Chrom: chrom_del, ChromStart: mafRecords[i].Species[k].ELine.Start, ChromEnd: mafRecords[i].Species[k].ELine.Start + mafRecords[i].Species[k].ELine.Size, Name: "del", Score: int64(mafRecords[i].Score)} //get chrom,start,end,name,score
+						current_ins := bed.Bed{Chrom: chrom_ins, ChromStart: mafRecords[i].Species[k-1].SLine.Start, ChromEnd: mafRecords[i].Species[k-1].SLine.Start + mafRecords[i].Species[k-1].SLine.Size, Name: "ins", Score: int64(mafRecords[i].Score)}
+						bedList_del = append(bedList_del, &current_del) //append to growing bed
+						bedList_ins = append(bedList_ins, &current_ins)
+					}
+				}
+			}
+		}
+	}
+
+	//write out bed files
+	bed.Write(outBed_del, bedList_del, 5) //bed file has 5 fields
+	bed.Write(outBed_ins, bedList_ins, 5)
+}
+
+func usage() {
+	fmt.Print(
+		"humanInsertion - takes pairwise alignment maf and finds insertions in species_ins not present in species_del but flanked by continuous alignments\n" +
```

Comment:
Great job making this program more general than you need for your project.  i would also think about giving it a more general name since it can find insertions in all sorts of species (and also deletions).  Maybe "mafIndels"?